rtos/Mutex.h

```diff
@@ -54,13 +54,13 @@ typedef mbed::ScopedLock<Mutex> ScopedMutexLock;
  */
  
 /** The Mutex class is used to synchronize the execution of threads.
- This is for example used to protect access to a shared resource.
+ This is, for example, used to protect access to a shared resource.
 
  @note You cannot use member functions of this class in ISR context. If you require Mutex functionality within
  ISR handler, consider using @a Semaphore.
 
  @note
- Memory considerations: The mutex control structures will be created on current thread's stack, both for the Mbed OS
+ Memory considerations: The mutex control structures are created on the current thread's stack, both for the Mbed OS
```

Comment:
Wouldn't have noticed this note apart from the edit, but it's false.

The RTOS mutex control structure is embedded in the `Mutex` object itself (rather than any RTOS memory pool), so the allocation is determined by where you allocate the `Mutex`. That *could* conceivably be on the stack, but not in any normal application.

Reply:
@aashishc1988 @AnotherButler

Reply:
Sorry I was not in office for couple of days. We would need to file separate PR for it since this is already merged. @kjbracey-arm , would you please file a PR with the accurate description? Thanks